### PR TITLE
Add client cert condition when cert is rotated

### PR DESCRIPTION
**Pull request by qiujian16**
Signed-off-by: Jian Qiu <jqiu@redhat.com>

### CODE REVIEW

File: pkg/spoke/managedcluster/registration.go [hunk 1]
```diff
@@ -83,11 +87,29 @@ func NewClientCertForHubController(
 		spokeSecretInformer,
 		spokeKubeClient,
 		hubKubeClient,
+		statusUpdater,
 		recorder,
 		controllerName,
 	)
 }
 
+func GenerateBootStrapStatusUpdater() clientcert.StatusUpdateFunc {
```
Comment (elgnay) [inline pkg/spoke/managedcluster/registration.go]: nit: GenerateBootStrapStatusUpdater -> GenerateBootstrapStatusUpdater?

File: pkg/clientcert/cert_controller.go [hunk 2]
```diff
@@ -258,8 +285,43 @@ func (c *clientCertificateController) sync(ctx context.Context, syncCtx factory.
...
 			return err
 		}
+
+		timeBefore, timeAfter, err := getCertValidityPeriod(secret)
+
+		cond := metav1.Condition{
+			Type:    "ClusterCertificateRotated",
+			Status:  metav1.ConditionTrue,
+			Reason:  "ClientCertificateUpdated",
+			Message: fmt.Sprintf("client certificate rotated starting from %v to %v", *timeAfter, *timeBefore),
```
Comment (elgnay) [inline pkg/clientcert/cert_controller.go]: Need to swap timeAfter and timeBefore in the argument list.

Comment (qiujian16) [reply pkg/clientcert/cert_controller.go]: i thought timeAfter means cert is valid after timeAfter?

Comment (elgnay) [reply pkg/clientcert/cert_controller.go]: I checked the code one more time and found `getCertValidityPeriod(secret)` returns `notBefore`, `notAfter` and `error`. So the validity period of a cert should start from `notBefore` to `notAfter`. What do you think?

**Pull request merged**